pallets/democracy-mbm/src/lib.rs

```diff
@@ -339,14 +344,29 @@ impl<T: pallet_democracy::Config + frame_system::Config, W: weights::WeightInfo>
     }
 }
 
+pub struct DemocracyMigrationSaveMigrationBlock<T>(PhantomData<T>);
```

Comment:
This approach is good 👍 

One minor note, about the weight, I believe block number is whitelisted, so it's actually just 1 write. Not important though.

Reply:
Isn't this line a read ?
`        let block_number = frame_system::Pallet::<T>::block_number().saturated_into::<u32>();`

Reply:
It is, but it's whitelisted AFAIK

Reply:
I see now. Yes it's whitelisted. So I will update to `1 write` only